### ISSUE TITLE
Fail in line 861

### ISSUE THREAD

**Issue by romoromor** [zbx-sadman/unifi_miner#11]
Hi, i have Unifi version 5.6.22.
Your script works great, but when i get for example firmware version of AP, i get three lines of
Use of uninitialized value in string eq at /etc/zabbix/scripts/unifi_miner.pl line 861.
and then is firmware version numbers.
I'm not a python programmer, that mean, i don't know, what line 861 does.
For this moment i fix this issue by delete
, last if (@{$jsonData}[$i]->{$idKey} eq $_[3]);
from line 861.
But i think, that isn't right solution:-)
Please, can you help me?
Thanks

**Comment by zbx-sadman** [comment]
I need to see JSON, which caused this error and command which you use.

File which contained JSON are placed in temporary catalog (_/dev/shm_ on default) - it named like that: _https\_\_\_localhost\_8443\_api\_s\_default\_stat\_device\_00\_27\_22\_0a\_16\_a6\_

You can send it to my email (placed on my profile).

**Comment by martinonicolas** [comment]
I have the same problem with Unifi Controller 5.6.26 I check the firmware version of various UAPs with de command:
`/opt/unifi_miner/unifi_miner.pl -a "get" -o "uap" -s "default" -k "serial" -i "UAP_ID"  -c "0" -l "https://127.0.0.1:8443" -u 'xxxxxxxx' -p 'xxxxxxx'
Use of uninitialized value in string eq at /opt/unifi_miner/unifi_miner.pl line 861.
802AA8028071`

In some the result is ok, showing only the firmware version, but in others shows the mentioned error and then de firmware version.

Thanks!

**Comment by zbx-sadman** [comment]
Yes, this problem caused when a few unadopted devices exists on the UniFi Controller's Site. Seems that unadopted devices haven't "_id" key (may be since 5.5 branch started). If object that specified by -i  option will be found in the list before object without "_id" key - all ok, otherwise - error is caused.

It was fixed, but in [experimental](https://github.com/zbx-sadman/unifi_miner/blob/master/experimental/unifi_miner.pl) release. Script must work ok, just i make long test with Zabbix 3.4 features.

**Comment by rguy2001-zz** [comment]
HI thanks for sharing. Do you have zabbix template that I can upload fro unifi aps?

**Comment by zbx-sadman** [comment]
Use this tempates: https://github.com/zbx-sadman/unifi_proxy/tree/master/Zabbix_Templates
